tasks/Task.cpp

```diff
@@ -301,13 +341,135 @@ base::samples::RigidBodyState Task::convertToRbs(vpHomogeneousMatrix pose)
     rbs.position[0] = pose.getTranslationVector()[0];
     rbs.position[1] = pose.getTranslationVector()[1];
     rbs.position[2] = pose.getTranslationVector()[2];
-    
+
     vpRotationMatrix R;
     pose.extract(R);
-    
     vpQuaternionVector q(R);
-    
     rbs.orientation = base::Orientation(q.w(), q.x(), q.y(), q.z());
 
     return rbs;
 }
+
+void Task::transformInput(base::LinearAngular6DCommand cmd_in_body)
+{
+    //initialize setpoint on the body frame
+    //since the non-expected input are NaN, set it to default values
+    //in order to have valid values for the HomogeneusMatrix creation
+    for (int i = 0; i < 3; ++i)
+    {
+        if (!expected_inputs.linear[i]) {cmd_in_body.linear[i] = 0;}
+        if (!expected_inputs.angular[i])
+        {
+            if ( i == 0 || i == 1)
+            {
+                cmd_in_body.angular[i] = -M_PI_2;
```

Comment:
>  However .... This -M_PI_2 according to your comment is basically applying the frame-to-body transformation. Which you apply further down. So I still don't get it.

The -M_PI_2 indeed is related to the frame-to-body transformation, however the loop is just to assing real values to the ones that are NaN (since they are not expected inputs). This is done in order to allow the Homogeneus Matrix creation.